TestProject.BL.Test/Services/RatingServiceTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using TestProject.BL.Mappers;
+using TestProject.BL.Models;
+using TestProject.BL.Services;
+using TestProject.DAL.Enums;
+using TestProject.DAL.Models;
+using TestProject.DAL.Repositories;
+using Xunit;
+
+namespace TestProject.BL.Test.Services
+{
+    public class RatingServiceTests
+    {
+        private RatingService _ratingService;
+        private Mock<IRepository<Rating>> _mockRatingRepository;
+        private Mock<IRepository<User>> _mockUserRepository;
+        private Mock<IMapper<RatingModel, Rating>> _mockRatingMapper;
+
+        public RatingServiceTests()
+        {
+            _mockRatingRepository = new Mock<IRepository<Rating>>();
+            _mockUserRepository = new Mock<IRepository<User>>();
+            _mockRatingMapper = new Mock<IMapper<RatingModel, Rating>>();
+            _ratingService = new RatingService(_mockRatingRepository.Object, 
+                _mockUserRepository.Object, _mockRatingMapper.Object);
+        }
+
+        [Fact]
+        public async Task Should_set_rating()
+        {
+            var ratingModel = new RatingModel();
+            var rating = new Rating();
+            var user = new User();
+            _mockUserRepository.Setup(repo => repo.Get(It.IsAny<Func<User, bool>>())).Returns(new List<User>() { user });
+            _mockRatingMapper.Setup(mapper => mapper.ToDalModel(ratingModel)).Returns(rating);
+
+            await _ratingService.Set(ratingModel, "");
+
+            _mockRatingRepository.Verify(repo => repo.Create(rating), Times.Exactly(1));
+        }
+
+        [Fact]
+        public void When_rated_should_return_true()
+        {
+            var rating = new Rating();
+            var user = new User();
+            _mockUserRepository.Setup(repo => repo.Get(It.IsAny<Func<User, bool>>())).Returns(new List<User>() { user });
+            _mockRatingRepository.Setup(repo => repo.Get(It.IsAny<Func<Rating, bool>>())).Returns(new List<Rating>() { rating });
```

Comment:
You can use exact values here. This way you will verify that the values passed to the method are used correctly.

Reply:
And return a result according to the value and check it in the assert.